app/controllers/providers/home_address/non_uk_home_addresses_controller.rb

```diff
@@ -0,0 +1,28 @@
+module Providers
+  module HomeAddress
+    class NonUkHomeAddressesController < ProviderBaseController
+      def show
+        @form = Addresses::NonUkHomeAddressForm.new(model: non_uk_home_address)
+      end
+
+      def update
+        @form = Addresses::NonUkHomeAddressForm.new(form_params)
+        render :show unless save_continue_or_draft(@form)
+      end
+
+    private
+
+      def non_uk_home_address
+        applicant.address || applicant.build_address(location: "home")
```

Comment:
@skdejong I think this should be applicant.home_address not applicant.address however that won't work until ap-4833 is merged.